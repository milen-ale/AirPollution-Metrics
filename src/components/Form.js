import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import './allCss.modules.css';
import { fetchWeatherWithInput } from '../redux/weather/weatherReducer';

function Form() {
  const dispatch = useDispatch();

  const navigate = useNavigate();

  const handleSubmit = (lat, long) => {
    dispatch(fetchWeatherWithInput(lat, long));
    navigate(`/country/${lat}&${long}`);
  };

  const clear = () => {};

  return (
    <div className="form-container">
      {/* <div className="image">
        <img className="" src={globe} alt="" />
      </div> */}
      {/* <h2 className="text-3xl font-bold">Global Air Pollution Index</h2> */}
      <div className="text-center py-4">
        <form
          onSubmit={(e) => {
            e.preventDefault();
            handleSubmit(e.target[0].value, e.target[1].value);
            clear(e.target[0].value = '', e.target[1].value = '');
          }}
        >
          <label htmlFor="latitude">
            <p className="text">Latitude</p>
            <input
              className=""
              name="lat"
              id="latitude"
              type="number"
              placeholder="Enter a Latitude..."
              onChange={(e) => e.target.value}
              required
            />
            <br />
          </label>
          <label htmlFor="longitude">
            <p className="text">Longitude</p>
            <input
              className=""
              name="long"
              id="longitude"
              type="number"
              placeholder="Enter a Longitude..."
              onChange={(e) => e.target.value}
              required
            />
            <br />
          </label>
          <input
            className="search-btn"
            type="submit"
            value="Search"
          />
        </form>
      </div>
    </div>
  );
}

export default Form;
