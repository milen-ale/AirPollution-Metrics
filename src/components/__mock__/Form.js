/* eslint-disable */
import React from 'react';


const Form = () => (
  <div className="form-container">
  
    <div className="text-center py-4">
      <form
        onSubmit={(e) => {
          e.preventDefault();
          handleSubmit(e.target[0].value, e.target[1].value);
          clear(e.target[0].value = '', e.target[1].value = '');
        }}
      >
        <label htmlFor="latitude">
          <p className="text">Latitude</p>
          <input
            className=""
            name="lat"
            id="latitude"
            type="number"
            placeholder="Enter a Latitude..."
            onChange={(e) => e.target.value}
            required
          />
          <br />
        </label>
        <label htmlFor="longitude">
          <p className="text">Longitude</p>
          <input
            className=""
            name="long"
            id="longitude"
            type="number"
            placeholder="Enter a Longitude..."
            onChange={(e) => e.target.value}
            required
          />
          <br />
        </label>
        <input
          className="search-btn"
          type="submit"
          value="Search"
        />
      </form>
    </div>
  </div>
);

export default Form;
