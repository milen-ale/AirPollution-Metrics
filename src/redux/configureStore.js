import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import logger from 'redux-logger';
import weatherReducer from './weather/weatherReducer';

const reducer = combineReducers({
  weatherReducer,
});

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(logger, thunk)),
);

export default store;
