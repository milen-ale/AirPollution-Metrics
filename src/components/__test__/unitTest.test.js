import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Form from '../__mock__/Form';
import TopBar from '../Header';

describe('Components snapshot tests', () => {
  const mockStore = configureStore();

  it('Form page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <Form />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TopBar renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <TopBar />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
