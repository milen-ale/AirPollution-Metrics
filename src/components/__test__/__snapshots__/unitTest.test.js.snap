// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot tests Form page renders correctly 1`] = `
<div
  className="form-container"
>
  <div
    className="text-center py-4"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="latitude"
      >
        <p
          className="text"
        >
          Latitude
        </p>
        <input
          className=""
          id="latitude"
          name="lat"
          onChange={[Function]}
          placeholder="Enter a Latitude..."
          required={true}
          type="number"
        />
        <br />
      </label>
      <label
        htmlFor="longitude"
      >
        <p
          className="text"
        >
          Longitude
        </p>
        <input
          className=""
          id="longitude"
          name="long"
          onChange={[Function]}
          placeholder="Enter a Longitude..."
          required={true}
          type="number"
        />
        <br />
      </label>
      <input
        className="search-btn"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;

exports[`Components snapshot tests TopBar renders correctly 1`] = `
<nav
  className=""
>
  <div
    className="header-h1"
  >
    <h1>
      Global Air Pollution Metrics
    </h1>
  </div>
</nav>
`;
